src/extension/content-script/batteries/YouTubeVideo.ts

```diff
@@ -2,64 +2,89 @@ import getOriginData from "../originData";
 import { findLnurlFromYouTubeAboutPage } from "./YouTubeChannel";
 import { findLightningAddressInText, setLightningData } from "./helpers";
 
+declare global {
+  interface Window {
+    LBE_YOUTUBE_MUTATION_OBSERVER: MutationObserver;
+  }
+}
+
 const urlMatcher = /^https:\/\/www\.youtube.com\/watch.*/;
 
 const battery = async (): Promise<void> => {
-  let text = "";
-  document
-    .querySelectorAll(
-      "ytd-watch-metadata #above-the-fold #bottom-row #description #description-inner #description-inline-expander yt-attributed-string .yt-core-attributed-string"
-    )
-    .forEach((e) => {
-      text += ` ${e.textContent}`;
-    });
-  const channelLink = document.querySelector<HTMLAnchorElement>(
-    "ytd-watch-metadata #above-the-fold #top-row ytd-video-owner-renderer .ytd-video-owner-renderer .ytd-channel-name a"
-  );
-  if (!text || !channelLink) {
-    return;
-  }
-  let match;
-  let lnurl;
-  // check for an lnurl
-  if ((match = text.match(/(lnurlp:)(\S+)/i))) {
-    lnurl = match[2];
-  }
-  // if there is no lnurl we check for a zap emoji with a lightning address
-  // we check for the @-sign to try to limit the possibility to match some invalid text (e.g. random emoji usage)
-  else if ((match = findLightningAddressInText(text))) {
-    lnurl = match;
-  } else {
-    // load the about page to check for a lightning address
-    const match = channelLink.href.match(
-      /^https:\/\/www\.youtube.com\/(((channel|c)\/([^/]+))|(@[^/]+)).*/
+  async function youtubeDOMChanged(
```

Comment:
did we use some mutation observer before?

Reply:
yes we use it for twitter battery. as its required for twitter as well